src/client/opamAdminCommand.ml

```diff
@@ -331,6 +331,100 @@ let packages_with_prefixes repo_root packages =
             (List.rev missing_pkgs)));
   pkgs_map
 
+let add_extrafiles_command_doc =
+  "Add extra-files to an opam repository."
+let add_extrafiles_command cli =
+  let command = "add-extrafiles" in
+  let doc = add_extrafiles_command_doc in
+  let man = [
+    `S Manpage.s_description;
+    `P "This command scans through package definitions, and add extrafiles \
+        field as requested."
+  ]
+  in
+  let hash_kinds = [`MD5; `SHA256; `SHA512] in
+  let hash_type_arg =
+    OpamArg.mk_opt ~cli OpamArg.(cli_from cli2_2) ["hash"]
+      "HASH_ALGO" "The hash, or hashes to be added"
+      Arg.(some (enum
+                   (List.map (fun k -> OpamHash.string_of_kind k, k)
+                      hash_kinds))) None
+  in
+  let packages =
+    OpamArg.mk_opt ~cli OpamArg.(cli_from cli2_2) ["p";"packages"]
+      "PACKAGES" "Only add extra files for the given packages"
+      Arg.(list OpamArg.package) []
+  in
+  let cmd global_options hash_type packages () =
+    OpamArg.apply_global_options cli global_options;
+    let repo_root = checked_repo_root () in
+    let pkg_prefixes = packages_with_prefixes repo_root packages in
+    let compute ?kind file =
+      OpamHash.compute ?kind (OpamFilename.to_string file)
+    in
+    let has_error =
+      OpamPackage.Map.fold (fun nv prefix has_error ->
+          let opam_file = OpamRepositoryPath.opam repo_root prefix nv in
+          let opam = OpamFile.OPAM.read opam_file in
+          let has_error =
+            if OpamFile.exists (OpamRepositoryPath.url repo_root prefix nv) then
+              (OpamConsole.warning "Not updating external URL file at %s"
+                 (OpamFile.to_string
+                    (OpamRepositoryPath.url repo_root prefix nv));
+               true)
+            else has_error
+          in
+          let files_dir = OpamRepositoryPath.files repo_root prefix nv in
+          if OpamFilename.exists_dir files_dir then
+            (let files =
+               OpamFilename.rec_files files_dir
+               |> List.map (fun file ->
+                   file,
+                   OpamFilename.Base.of_string
+                     (OpamFilename.remove_prefix files_dir file))
+             in
+             let extra_files =
+               OpamStd.Option.default [] (OpamFile.OPAM.extra_files opam)
+             in
+             let extra_files = OpamFilename.Base.Map.of_list extra_files in
+             let xfiles =
+               List.fold_left (fun extra (file, base) ->
+                   match OpamFilename.Base.Map.find_opt base extra_files with
+                   | Some hash ->
+                     let kind = OpamHash.kind hash in
+                     let xhash = compute ~kind file in
+                     if OpamHash.compare hash xhash <> 0 then
```

Comment:
```suggestion
                     if not (OpamHash.equal hash xhash) then
```

Reply:
equal is already a comparison with 0. Or you propose this change for another purpose?